### DISCUSSION TITLE
facetDistribution being limited by distinctAttribute does not always make sense

### DISCUSSION THREAD

**Discussion by chamby** [meilisearch/product#741] [Feedback & Feature Proposal]
Right now `facetDistribution` is limited by the use of `distinctAttribute`. The implementation makes sense but is not ideal from a user perspective.

My use case matches Meilisearch's own [example](https://www.meilisearch.com/docs/learn/configuration/distinct#example).

I only want to show distinct results based on a single graphic design but with many variants of color, material, and cut represented within that design. The faceted attributes will decide which result for each design is the most relevant. The following screenshot shows many color attributes in the result set.

<img width="685" alt="many design results with distinctAttribute enabled" src="https://github.com/meilisearch/product/assets/725790/982fc07b-58e0-44f9-9756-7b9ad175d1f7">

If I narrow the search to precisely one distinct result the problem becomes more obvious. The `facetDistribution` is built from the distinct results ignoring that there are still many valid/possible results within the distinct design group. Other colors are still relevant for the color facet to report, but the `distinctAttribute` prevents this.

<img width="681" alt="single design result with distinctAttribute enabled" src="https://github.com/meilisearch/product/assets/725790/212b470f-8151-4c02-a008-507631b18fe9">

Here are the results with the same search criteria when I disable the `distinctAttribute`. Now the facets show what I expect.

<img width="703" alt="single design result with no distinctAttribute disabled" src="https://github.com/meilisearch/product/assets/725790/9d4e4f25-9cbc-40ce-a864-c3ca65c17f5e">

Would it be possible for Meilisearch to return a `facetDistribution` that ignores `distinctAttribute`? A user should be able to select purple and see the single result swap from red to purple as the relevancy changes.

**Comment by Kerollmops** [comment]
Hey @chamby 👋

Thank you very much for the report. Unfortunately, the `disctinctAttributes` is an index setting and is therefore _always applied_. We should introduce a new `distinguishableAttributes` akin to the `filterableAttributes` in the sense that it can be set or not at query time. This way you could decide on which attribute(s) you would like to distinct on or not in the facets or on the results.

What do you think about this @meilisearch/product-team? Note that the distinct system is already computed at query time, the main complexity is the setting and API.

**Comment by chamby** [comment]
@Kerollmops Thanks for implementing support for this in https://github.com/meilisearch/meilisearch/pull/4693 released in v1.9. Is there a straightforward way to have this search parameter set for the results but not the facets in instant-meilisearch? From a brief review [meiliSearchParams](https://github.com/meilisearch/meilisearch-js-plugins/tree/main/packages/instant-meilisearch#meilisearch-search-parameters) seems to apply to all requests.

**Reply by Kerollmops** [reply]
Hey 👋

> Is there a straightforward way to have this search parameter set for the results but not the facets in instant-meilisearch?

The best way is to do two independent search requests with and without the constraints. You can do that in a single HTTP call with the multi-search endpoint.

Have a nice day!
